packages/vscode-extension/src/codeLensProvider.ts

```diff
@@ -411,7 +411,12 @@ export class AadAppTemplateCodeLensProvider implements vscode.CodeLensProvider {
   public provideCodeLenses(
     document: vscode.TextDocument
   ): vscode.ProviderResult<vscode.CodeLens[]> {
-    if (document.fileName.endsWith("template.json")) {
+    // V3 supports customize aad manifest
+    if (
+      isV3Enabled()
+        ? document.getText().includes("${{")
```

Comment:
I'm not sure whether this will cause performance issue for large document opened by users, and maybe this might activate code lense for invalid document (like a markdown that contains `${{`)?